src/SmoFacade/Server.cs

```diff
@@ -192,38 +192,63 @@ public Database Database(string dbName)
 
       var restore = new Restore { Database = databaseName, NoRecovery = true };
 
-      foreach(var backup in backupOrder) {
-        var device = BackupFileTools.IsValidFileUrl(backup.PhysicalDeviceName) ? DeviceType.Url : DeviceType.File;
-        var backupDeviceItem = new BackupDeviceItem(backup.PhysicalDeviceName, device);
-        if(_credentialName != null && device == DeviceType.Url)
-          backupDeviceItem.CredentialName = _credentialName;
+      foreach(var stripedBackupSet in stripedBackupSetChain) {
 
-        restore.Devices.Add(backupDeviceItem);
+        AddBackupDeviceItemsToRestore(restore, stripedBackupSet);
 
         var defaultFileLocations = DefaultFileLocations();
-        if(defaultFileLocations != null) {
-          restore.RelocateFiles.Clear();
-          var fileList = policy.Execute(() => restore.ReadFileList(_server).AsEnumerable());
-          foreach(var file in fileList) {
-            var physicalName = (string)file["PhysicalName"];
-            var fileName = GetFileName(physicalName);
+        if (defaultFileLocations != null)

```

Comment:
Looks like this query actually only needs to be done 1 time, not on each loop iteration.
Can you move it out of the loop?
https://github.com/factset/AgDatabaseMove/blob/a89418883f5c3275b561b6edd943f6f0bf76dbed/src/SmoFacade/Server.cs#L123